cmake/onnxruntime_providers.cmake

```diff
@@ -471,7 +471,9 @@ if (onnxruntime_USE_CUDA)
   onnxruntime_add_include_to_target(onnxruntime_providers_cuda onnxruntime_common onnxruntime_framework onnx onnx_proto ${PROTOBUF_LIB} flatbuffers::flatbuffers)
   if (onnxruntime_ENABLE_TRAINING_OPS)
     onnxruntime_add_include_to_target(onnxruntime_providers_cuda onnxruntime_training)
-    target_link_libraries(onnxruntime_providers_cuda PRIVATE onnxruntime_training)
+    if (onnxruntime_ENABLE_TRAINING)
```

Comment:
if "if (onnxruntime_ENABLE_TRAINING_OPS)" is true, then "if (onnxruntime_ENABLE_TRAINING)" is ture, right?

Reply:
not always... 
1. On Device Training build uses enable_training_apis flag and in this case enable_training will not be true
2. There is 1 more scenario where training ops are included in the inference build... I am not 100% sure how it is used but this enable_trianing_ops macro was first added for this scenario.